contracts/plugins/trading/DutchTrade.sol

```diff
@@ -283,7 +321,15 @@ contract DutchTrade is ITrade, Versioned {
         returns (uint256 soldAmt, uint256 boughtAmt)
     {
         require(msg.sender == address(origin), "only origin can settle");
-        require(bidder != address(0) || block.timestamp > endTime, "auction not over");
+        require(
+            bidder != address(0) ||
+                address(activeTrustedFill) != address(0) ||
+                block.timestamp > endTime,
+            "auction not over"
+        );
+
+        // Close any active trusted fill and reclaim tokens
+        _closeTrustedFill();
```

Comment:
We can't confirm the buyer transferred any tokens through the filler, right? Anyone can call `trading.settleTrade(..)` to deny the auction. I think there aren't any other consequences other than denying it, but still, we need a slightly different design here (unless I'm missing something).

Reply:
Looking at it now, I think the `require` here is wrong and we should probably enforce reaching the auction end time in the case of Trusted fillers.

Reply:
Modified `settleTrade` to only settle the ongoing auction if we have the `buy` tokens we require from the filler.